nibabel/spatialimages.py

```diff
@@ -325,6 +326,21 @@ class SpatialImage(DataobjImage):
     ''' Template class for volumetric (3D/4D) images '''
     header_class = SpatialHeader
 
+    class Slicer(object):
```

Comment:
Remind me why the Slicer class is internal to the SpatialImage class?  I think that will mean that you can no longer pickle images.

Reply:
The reason is that we wanted something that could be used like:

```Python
img.slicer[sliceobj]
```

Some other way of organizing the code would be fine with me.

Reply:
I think it would make pickle happy if the class definition was at the top level, like:

```
class ImageSlicer(object):
```

Does that work?

Reply:
Sure. In which case we definitely shouldn't use a private method like `img._slice_affine`. Does it make more sense to place it here:

```Python
class ImageSlicer(object):
    ...

    @classmethod
    def slice_affine(affine, slicer):
        ...
```

I'm mostly unsure how much sense `img.slice_affine` as part of the image API.